src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -48,9 +48,11 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _sourceRepositoryProvider = sourceRepositoryProvider;
             _settings = settings;
             _solutionManager = solutionManager;
-            _projectContext = new VSAPIProjectContext();
             _packageServices = packageServices;
             _deleteOnRestartManager = deleteOnRestartManager;
+
+            _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
```

Comment:
If it is `VSAPIProjectContext` that assumes it is running on the UI thread (or uses RPC otherwise), then this change may simply move the deadlock from MEF part activation to a later time if that `Lazy<T>` is ever evaluated on a background thread.

If instead of making it `Lazy<T>` you make it `AsyncLazy<T>` then explicitly switch to the main thread before instantiating this type, you'll have to update synchronous users of this Lazy as well to use JTF.Run but you'll avoid another set of deadlocks. You'll then be vulnerable to anyone that blocks the UI thread in a non JTF wait and yet needs your background thread to complete, but you can't win 'em all, and that hopefully is not happening or happening much less than the problem you're trying to fix.